src/execution/IncrementalPublisher.ts

```diff
@@ -609,27 +609,31 @@ export class IncrementalPublisher {
     deferredGroupedFieldSetRecord: DeferredGroupedFieldSetRecord,
   ): IncrementalDeferResult {
     const { data, deferredFragmentRecords } = deferredGroupedFieldSetRecord;
-    let maxLength = deferredFragmentRecords[0].path.length;
-    let maxIndex = 0;
-    for (let i = 1; i < deferredFragmentRecords.length; i++) {
-      const deferredFragmentRecord = deferredFragmentRecords[i];
+    let maxLength: number | undefined;
+    let recordWithLongestPath: DeferredFragmentRecord | undefined;
+    for (const deferredFragmentRecord of deferredFragmentRecords) {
+      if (deferredFragmentRecord.id === undefined) {
+        continue;
+      }
       const length = deferredFragmentRecord.path.length;
-      if (length > maxLength) {
+      if (maxLength === undefined || length > maxLength) {
         maxLength = length;
-        maxIndex = i;
+        recordWithLongestPath = deferredFragmentRecord;
```

Comment:
You can remove one `!` by declaring `id` before the loop and assigning `id` inside this condition.

Reply:
awesome, thanks

[969f417](https://github.com/graphql/graphql-js/pull/3987/commits/969f4173dd7b241c885c5920fd4e7b23961e1e0a)